### PR TITLE
Add minimum client version to feature activation

### PR DESCRIPTION
**Pull request by zathras-crypto**
Hopefully uncontroversial change to feature activations.

Basically we need to nominate a minimum supported client version within the activation message, as clients may be aware of a feature before they fully support the correct implementation.

For example:

Client 0.10.x knows about feature id 2 (betting) and has no implementation
Client 0.11.0 knows about feature id 2 (betting) and has basic implementation
Client 0.11.1 knows about feature id 2 (betting) and has an improved implementation
Client 0.11.2 knows about feature id 2 (betting) and has some bug fixes to betting
Client 0.11.3 knows about feature id 2 (betting) and has the "final" implementation prior to betting go live

If we just activate as per current, all these clients will go active and have differing interpretations of the state due to differening betting code (ie, out of consensus).

The only way I can see to address this is to embed an Omni Core version in the activation, such that (in the example) only 0.11.3+ clients would be activated and older would be forced to update.

I've done this as a standalone PR in case there are objections, but really it's to go hand in hand with the work I'm doing on alerts as feature activation will integrate there.

### CODE REVIEW

**Comment by dexX7** [comment]
Ah, this seems like a good idea, but I'm not yet sure, how this fits together with the alerting system.

I feel like it mixes responsibilities ("which code part does what", not real life policy), because isn't this actually something that should be covered by the alerting system? But still, the alerting system has no concept of features (like class C), but only transaction types, so point on that.

At the same time it feels a bit like sub actions: instead of [feature identifier], features are marked with [feature identifier, client version]. Couldn't we just increment the feature identifier? This is a concept question, and I'd prefer to use the client version here, too.

ACK on the general idea, however, I'd like to know what happens with the alerting system them: it seems to render at least the alert part about transaction types unnecessary, unless I misunderstand it?

**Comment by dexX7** [comment]
Alright, after taking another look at #169, it looks like the alert-for-transaction-types no longer exists.

**Comment by zathras-crypto** [comment]
> how this fits together with the alerting system.
> the alerting system has no concept of features (like class C)
> it looks like the alert-for-transaction-types no longer exists.

Sorry, perhaps I shouldn't have split this off and should have kept it together with the alerting stuff.

So a few thoughts of mine in no particular order...

* I found alerting was incapable of being used to notify about features that didn't explicitly make a new transaction type live
* I found we needed the ability to have multiple alerts active
* I found the old type 4 tx/ver type check didn't fit well with feature activation, as things needed to be done in a certain order (eg activate message has to be before alert type 4 message) and we still had no granular control over client version in an old type 4 tx/ver check 
* I figured the best thing to do would be to integrate feature activations with the alerting system
* I intend to replace the tx/ver auto shutdown check with a min client version feature activation
* A new feature activation message would trigger an alert
* A feature going live would trigger an alert
* A non-supported client would auto-shutdown on unsupported feature live block - a replacement of the transaction tx/ver type check auto-shutdown (old alert type 4)
 
Basically in a nutshell the idea is that we broadcast a feature activation message, and that fires appropriate alerts automatically on activation & on live.

Should I just move this stuff into the alerts PR and put it all together in one place?

> Couldn't we just increment the feature identifier?

We could I guess, but it kind of feels like a workaround and it's another thing we need to remember to increment each build etc.  We already have a client version, so it seemed logical to use that attribute.

**Comment by dexX7** [comment]
> Should I just move this stuff into the alerts PR and put it all together in one place?

Sure, if you think this is the way to go please go ahead. :)

**Comment by zathras-crypto** [comment]
Continued in #175
